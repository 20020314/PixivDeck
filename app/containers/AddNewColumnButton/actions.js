// @flow
import type { User } from '../../types/user'
import type { Action } from './actionTypes.js'
import { ADD_USER_ILLUST } from './constants'

export function addUserIllust(user: User): Action {
  return {
    type: ADD_USER_ILLUST,
    user,
  }
}
