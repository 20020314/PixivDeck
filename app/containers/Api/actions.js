// @flow
import type { Response } from 'services/api'

import { API_REQUEST_SUCCESS } from './actionTypes'
import type { ApiRequestSuccess } from './actionTypes'

export function apiRequestSuccess(response: Response): ApiRequestSuccess {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
