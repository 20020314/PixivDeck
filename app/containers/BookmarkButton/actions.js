// @flow
import type { Restrict } from './types'

import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILURE,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_FAILURE,
} from './actionTypes'
import type {
  AddBookmarkRequest,
  AddBookmarkSuccess,
  AddBookmarkFailure,
  DeleteBookmarkRequest,
  DeleteBookmarkSuccess,
  DeleteBookmarkFailure,
} from './actionTypes'

export function addBookmarkRequest(
  id: number,
  restrict: Restrict
): AddBookmarkRequest {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}
export function addBookmarkSuccess(
  id: number,
  restrict: Restrict
): AddBookmarkSuccess {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}
export function addBookmarkFailure(
  id: number,
  error: string
): AddBookmarkFailure {
  return {
    type: ADD_BOOKMARK_FAILURE,
    id,
    error,
  }
}
export function deleteBookmarkRequest(id: number): DeleteBookmarkRequest {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}
export function deleteBookmarkSuccess(id: number): DeleteBookmarkSuccess {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}
export function deleteBookmarkFailure(
  id: number,
  error: string
): DeleteBookmarkFailure {
  return {
    type: DELETE_BOOKMARK_FAILURE,
    id,
    error,
  }
}
