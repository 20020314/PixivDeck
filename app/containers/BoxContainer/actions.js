// @flow
import type { Action } from './actionTypes.js'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
