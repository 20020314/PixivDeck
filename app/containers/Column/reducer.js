// @flow
import update from 'utils/update'

export type BaseColumn = {
  illustIds: Array<number>,
  nextUrl: ?string,
}

type ActionTypes = {
  ADD_COLUMN_SUCCESS: $Subtype<string>,
  SET_NEXT_URL: $Subtype<string>,
}

export function baseReducer<
  T: BaseColumn,
  S: { [*]: T },
  AS: ActionTypes,
  A: { +type: $Subtype<string> }
>(types: AS, state: S, action: A): S {
  switch (action.type) {
    case types.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })
    case types.ADD_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    default:
      return state
  }
}
