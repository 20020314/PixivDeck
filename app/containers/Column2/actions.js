// @flow
import type { Action } from './actionTypes.js'
import { ADD_COLUMN } from './constants'
import type { Body } from './reducer'

export function addColumn(id: string, body: Body): Action {
  return {
    type: ADD_COLUMN,
    id,
    body,
  }
}
