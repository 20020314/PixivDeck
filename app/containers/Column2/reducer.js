// @flow
import type { Action } from './actionTypes'
import * as Actions from './constants'

type ColumnId = string
type BodyId = string
type BodyType = 'RANKING' | 'BOOKMARK'

export type Column = {
  id: ColumnId,
  title: string,
  bodyId: BodyId,
  type: BodyType,
}

export type Body = {
  id: BodyId,
  type: BodyType,
}

export type Ranking = {
  endpoint: '/v1/illust/ranking',
  params: {
    mode: string,
    offset: number,
  },
} & Body

export type State = {
  [id: string]: Column,
}

const initialState: State = {}

export default function(
  state: State = initialState,
  action: Action
): $Shape<State> {
  switch (action.type) {
    case Actions.ADD_COLUMN:
      return { ...state, [action.id]: action.body }
    default:
      return state
  }
}
