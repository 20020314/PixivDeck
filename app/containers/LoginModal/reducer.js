// @flow
import type { Action } from './actionTypes'
import * as Actions from './constants'

export type State = {
  username: string,
  password: string,
  isLoading: boolean,
  isLoginFailure: boolean,
}

export const initialState: State = {
  username: '',
  password: '',
  isLoading: false,
  isLoginFailure: false,
}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.SET_AUTH:
      return { ...state, username: action.username, password: action.password }
    case Actions.LOGIN_REQUEST:
      return { ...state, isLoading: true }
    case Actions.START_LOADING:
      return { ...state, isLoading: true }
    case Actions.END_LOADING:
      return { ...state, isLoading: false }
    case Actions.LOGIN_FAILURE:
      return { ...state, isLoginFailure: true }
    case Actions.CLEAR_ERROR:
      return { ...state, isLoginFailure: false }
    default:
      return state
  }
}
