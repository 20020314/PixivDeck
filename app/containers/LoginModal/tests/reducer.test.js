// @flow
import reducer, { initialState } from '../reducer'
import * as actions from '../actions'

test('default action', () => {
  // $FlowFixMe
  expect(reducer(undefined, { type: 'default action' })).toMatchSnapshot()
})

test('SET_AUTH', () => {
  expect(
    reducer(initialState, actions.setAuth('hello', 'world'))
  ).toMatchSnapshot()
})

test('LOGIN_REQUEST', () => {
  expect(
    reducer(initialState, actions.loginRequest('hello', 'world')).isLoading
  ).toMatchSnapshot()
})

test('LOGIN_FAILURE', () => {
  expect(reducer(initialState, actions.loginFailure())).toMatchSnapshot()
})

test('CLEAR_ERROR', () => {
  const store = { ...initialState, isLoginFailure: true }
  expect(reducer(store, actions.clearError())).toMatchSnapshot()
})
