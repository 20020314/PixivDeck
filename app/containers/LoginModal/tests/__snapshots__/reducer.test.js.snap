// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLEAR_ERROR 1`] = `
Object {
  "account": null,
  "isLoading": false,
  "isLoginFailure": false,
  "password": "",
  "username": "",
}
`;

exports[`LOGIN_FAILURE 1`] = `
Object {
  "account": null,
  "isLoading": false,
  "isLoginFailure": true,
  "password": "",
  "username": "",
}
`;

exports[`LOGIN_REQUEST 1`] = `true`;

exports[`SET_ACCOUNT 1`] = `
Object {
  "account": Object {
    "account": "akameco",
    "id": "19785907",
    "isMailAuthorized": true,
    "isPremium": false,
    "mailAddress": "fake@fake.com",
    "name": "akameco",
    "profileImageUrls": Object {
      "px16x16": "imgurl",
      "px170x170": "imgurl",
      "px50x50": "imgurl",
    },
    "xRestrict": 2,
  },
  "isLoading": false,
  "isLoginFailure": false,
  "password": "",
  "username": "",
}
`;

exports[`SET_AUTH 1`] = `
Object {
  "account": null,
  "isLoading": false,
  "isLoginFailure": false,
  "password": "world",
  "username": "hello",
}
`;

exports[`default action 1`] = `
Object {
  "account": null,
  "isLoading": false,
  "isLoginFailure": false,
  "password": "",
  "username": "",
}
`;
